script/tool/lib/src/publish_plugin_command.dart

```diff
@@ -110,50 +134,231 @@ class PublishPluginCommand extends PluginCommand {
+          continue;
+      }
+      _print('\n');
+      if (await _publishAndTagPackage(
+        packageDir: pubspecFile.parent,
+        remote: remote,
+        remoteUrl: remoteUrl,
+        shouldPushTag: shouldPushTag,
+      )) {
+        packagesReleased.add(pubspecFile.parent.basename);
+      } else {
+        packagesFailed.add(pubspecFile.parent.basename);
+      }
+      _print('\n');
+    }
+    if (packagesReleased.isNotEmpty) {
+      _print('Packages released: ${packagesReleased.join(', ')}');
+    }
+    if (packagesFailed.isNotEmpty) {
+      _print(
+          'Packages failed to release: ${packagesFailed.join(', ')}, see above for details.');
+    }
+    return packagesFailed.isEmpty;
+  }
 
-    final Directory packageDir = _getPackageDir(package);
-    await _publishPlugin(packageDir: packageDir);
+  // Publish the package to pub with `pub publish`.
+  // If `_tagReleaseOption` is on, git tag the release.
+  // If `shouldPushTag` is `true`, the tag will be pushed to `remote`.
+  // Returns `true` if publishing and tag are successful.
+  Future<bool> _publishAndTagPackage({
+    @required Directory packageDir,
+    @required String remote,
+    @required String remoteUrl,
+    @required bool shouldPushTag,
+  }) async {
+    if (!await _publishPlugin(packageDir: packageDir)) {
+      return false;
+    }
     if (argResults[_tagReleaseOption] as bool) {
-      await _tagRelease(
-          packageDir: packageDir,
-          remote: remote,
-          remoteUrl: remoteUrl,
-          shouldPushTag: shouldPushTag);
+      if (!await _tagRelease(
+        packageDir: packageDir,
+        remote: remote,
+        remoteUrl: remoteUrl,
+        shouldPushTag: shouldPushTag,
+      )) {
+        return false;
+      }
     }
-    await _finishSuccesfully();
+    _print('Released [${packageDir.basename}] successfully.');
+    return true;
   }
 
-  Future<void> _publishPlugin({@required Directory packageDir}) async {
-    await _checkGitStatus(packageDir);
-    await _publish(packageDir);
+  // Returns `true` if needs to release the version, `false` if needs to skip
+  Future<_CheckNeedsReleaseResult> _checkNeedsRelease({
+    @required File pubspecFile,
+    @required GitVersionFinder gitVersionFinder,
+    @required List<String> existingTags,
+  }) async {
+    if (!pubspecFile.existsSync()) {
+      _print('''
+The file at The pubspec file at ${pubspecFile.path} does not exist. Publishing will not happen for ${pubspecFile.parent.basename}.
+Safe to ignore if the package is deleted in this commit.
+''');
+      return _CheckNeedsReleaseResult.noRelease;
+    }
+
+    final Pubspec pubspec = Pubspec.parse(pubspecFile.readAsStringSync());
+    if (pubspec.publishTo == 'none') {
+      return _CheckNeedsReleaseResult.noRelease;
+    }
+
+    if (pubspec.version == null) {
+      _print('No version found. A package that '
+          'intentionally has no version should be marked '
+          '"publish_to: none".');
```

Comment:
Nit: Fix the excessive line breaking here?

Reply:
done